beancount-core/src/date.rs

```diff
@@ -0,0 +1,31 @@
+use std::borrow::Cow;
+
+#[cfg(feature = "chrono")]
+use chrono::NaiveDate;
+
+#[derive(Eq, PartialEq, Debug, Clone)]
+pub struct Date<'a> {
+    s: Cow<'a, str>
```

Comment:
I think this either should be public or there should be some other way of extracting the string from the `Date` struct.

I would also suggest using a tuple struct here instead, the field name is not very descriptive anyway?

Reply:
That's a good point. Isn't the parameter from a tuple struct public anyway?